teps/0060-remote-resource-resolution.md

```diff
@@ -0,0 +1,370 @@
+  proprietary internal catalog, or any other storage system.
+- `Remote Resource`: The YAML file or other representation of a Tekton resource
+  that lives in the Remote.
+- `Resolver`: A program or piece of code that knows how to interpret a
+  reference to a remote resource and fetch it for execution as part of the
+  user's TaskRun or Pipeline.
+- `Resource Resolution`: The act of taking a reference to a Tekton resource in
+  a Remote, fetching it, and returning its content to the Tekton Pipelines
+  reconcilers.
+
+## Motivation
+
+Pipelines currently provides support for resources to be run from two different
+locations: a Task or Pipeline can be run from the same cluster or (when the
+[`enable-tekton-oci-bundles`
+flag](https://github.com/tektoncd/pipeline/blob/main/config/config-feature-flags.yaml)
+is `"true"`) from an image registry hosting [Tekton
+Bundles](https://tekton.dev/docs/pipelines/pipelines/#tekton-bundles).
+
+This existing support has a few problems:
+
+1. For resources in a cluster the degree of control afforded to operators is
+   extremely narrow: Either a resource exists in the cluster or it does not. To
+   add new Tasks and Pipelines operators have to manually (or via automation)
+   sync them in (`kubectl apply -f git-clone.yaml`).
+2. For resources in a Tekton Bundle registry, operators don't have a choice
+   over what kind of storage best suits their teams: It's either a registry or
+   it's back to manually installing Tasks and Pipelines. This might not jive
+   with an org's chosen storage. Put another way: [why can't we just keep our
+   Tasks in git?](https://github.com/tektoncd/pipeline/issues/2298)
+3. Pipeline authors have to document the Tasks their Pipeline depends on
+   out-of-band. A [common example of
+   this](https://github.com/tektoncd/pipeline/discussions/3822) is a build
+   Pipeline that needs `git-clone` to be installed from the open source catalog
+   before the Pipeline can be run.
+   [TEP-0053](https://github.com/tektoncd/community/pull/352) is also exploring
+   ways of encoding this information as part of pipelines in the catalog.
+4. Tutorials introducing Tekton Pipelines have to include instructions for
+   fetching and manually installing tasks that the tutorial relies on.
+5. Pipelines' [resolver
+   code](https://github.com/tektoncd/pipeline/blob/bad45506a9a9c25b028703e85e501b8f97598695/pkg/reconciler/taskrun/resources/taskref.go#L38)
+   is synchronous: a reconciler thread is blocked while a resource is being
+   fetched.
+6. Pipelines' existing support for cluster and bundle resources is not easily
+   extensible without modifying Pipelines' source code.
+
+### Goals
+
+- Aspirationally, never require a user to run `kubectl apply -f git-clone.yaml`
+  ever again.
+- Provide the ability to manage Tekton resources in an org's source of choice
+  via any process of their choice - `git push`, `tkn bundle push`, `gsutil cp`
+  and so on, instead of `kubectl apply`.
```

Comment:
Note that we do not prevent that currently, we just do not cover it as part of tektoncd components, but users can create/use their own software to do this if they need.